File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/FarmTaskRecord.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FarmTaskRecord {

    private Long id;

    private Long taskId;

    private String taskTitle;

    private Long userId;

    private Long plotId;

    private String plotName;

    private Long operatorId;

    /** 操作人名称快照，避免用户改名后历史时间线失真。 */
    private String operatorNameSnapshot;

    private String operatorName;

    private Integer actionType;

    private String actionContent;

    private Integer resultStatus;

    /** 本次过程反馈的完成百分比，取值范围为 0-100。 */
    private Integer progressPercent;

    private Integer feedbackScore;

    private String feedbackDetail;

    private String optimizeSuggestion;

    /** 执行凭证附件地址的 JSON 数组，为后续图片上传能力预留。 */
    private String attachments;

    /** 客户端幂等请求号，防止弱网重试或连续点击重复生成事件。 */
    private String requestId;

    /** 事件来源：FARM_APP、SMART_FARM 或 SYSTEM。 */
    private String sourceClient;

    private Integer beforeStatus;

    private Integer afterStatus;

    private LocalDateTime executeTime;

    private LocalDateTime createTime;
}

File: smart_farm/src/api/farmTaskRecord.ts
import { get, http } from '@/utils/request'
import type { PageInfo } from '@/api/user'

export interface FarmTaskRecord {
  id: number
  taskId: number
  taskTitle?: string
  userId?: number
  plotId?: number
  plotName?: string
  operatorId?: number
  /** 操作时保存的人员名称快照。 */
  operatorNameSnapshot?: string
  operatorName?: string
  actionType: number
  actionContent?: string
  resultStatus?: number
  /** 阶段性进度，范围为0到100。 */
  progressPercent?: number
  feedbackScore?: number
  feedbackDetail?: string
  optimizeSuggestion?: string
  /** 附件地址JSON，为后续查看执行凭证预留。 */
  attachments?: string
  /** 客户端幂等请求号。 */
  requestId?: string
  /** 事件来源：FARM_APP、SMART_FARM 或 SYSTEM。 */
  sourceClient?: string
  beforeStatus?: number
  afterStatus?: number
  executeTime?: string
  createTime?: string
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

export const addFarmTaskRecord = (data: Partial<FarmTaskRecord>) => {
  return http<ApiResult<FarmTaskRecord>>({ url: '/farm-task-record/add', method: 'post', data })
}

export const getFarmTaskRecordById = (id: number) => {
  return get<ApiResult<FarmTaskRecord>>(`/farm-task-record/${id}`)
}

export const listFarmTaskRecords = (params: { taskId?: number; pageNum: number; pageSize: number }) => {
  return get<ApiResult<PageInfo<FarmTaskRecord>>>(
    '/farm-task-record/list',
    params as unknown as Record<string, unknown>,
  )
}
